test/server/http_test_server_filter_integration_test.cc

```diff
@@ -1,94 +1,38 @@
-#include "envoy/upstream/cluster_manager.h"
-#include "envoy/upstream/upstream.h"
-
-#include "external/envoy/test/common/upstream/utility.h"
-#include "external/envoy/test/integration/http_integration.h"
-
 #include "api/server/response_options.pb.h"
 #include "api/server/response_options.pb.validate.h"
 
 #include "server/configuration.h"
 #include "server/http_test_server_filter.h"
-#include "server/well_known_headers.h"
+
+#include "test/server/http_filter_integration_test_base.h"
 
 #include "gtest/gtest.h"
 
 namespace Nighthawk {
 
 using namespace testing;
-constexpr absl::string_view kBadJson = "bad_json";
 
-class HttpTestServerIntegrationTestBase : public Envoy::HttpIntegrationTest,
-                                          public TestWithParam<Envoy::Network::Address::IpVersion> {
-public:
-  HttpTestServerIntegrationTestBase()
-      : HttpIntegrationTest(Envoy::Http::CodecClient::Type::HTTP1, GetParam(), realTime()) {}
-
-  // TODO(oschaaf): Modify Envoy's Envoy::IntegrationUtil::makeSingleRequest() to allow for a way to
-  // manipulate the request headers before they get send. Then we can eliminate these copies.
-  Envoy::BufferingStreamDecoderPtr makeSingleRequest(
-      uint32_t port, absl::string_view method, absl::string_view url, absl::string_view body,
-      Envoy::Http::CodecClient::Type type, Envoy::Network::Address::IpVersion ip_version,
-      absl::string_view host, absl::string_view content_type,
-      const std::function<void(Envoy::Http::RequestHeaderMapImpl&)>& request_header_delegate) {
-    auto addr = Envoy::Network::Utility::resolveUrl(fmt::format(
-        "tcp://{}:{}", Envoy::Network::Test::getLoopbackAddressUrlString(ip_version), port));
-    return makeSingleRequest(addr, method, url, body, type, host, content_type,
-                             request_header_delegate);
-  }
+const std::string kDefaultProto = R"EOF(
```

Comment:
(optional) It is typically not recommended to create globals of object that have non-trivial destructors, since c++ doesn't guarantee or specify the order in which these objects get destructed. So it is very easy to end up in an undefined state where object A depends on B, but B gets destructed before A. We had a lot of crashes of servers while attempting proper shutdown due to this.

A good trick around it is to ensure that the global never gets destructed. One way of doing that is what the original code had, i.e. a constexpr absl::string_view instead of an std::string.

Doesn't really matter here, since this is a test file, so optional. It's just something that triggers my muscle memory.

Reply:
done in 2529ea6